pipeline/backend/ssh/ssh.go

```diff
@@ -0,0 +1,153 @@
+package ssh
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/melbahja/goph"
+
+	"github.com/woodpecker-ci/woodpecker/pipeline/backend/types"
+	"github.com/woodpecker-ci/woodpecker/shared/constant"
+)
+
+type ssh struct {
+	cmd        *goph.Cmd
+	output     io.ReadCloser
+	client     *goph.Client
+	workingdir string
+}
+
+type readCloser struct {
+	io.Reader
+}
+
+func (c readCloser) Close() error {
+	return nil
+}
+
+// make sure local implements Engine
+var _ types.Engine = &ssh{}
+
+// New returns a new ssh Engine.
+func New() types.Engine {
+	return &ssh{}
+}
+
+func (e *ssh) Name() string {
+	return "ssh"
+}
+
+func (e *ssh) IsAvailable() bool {
+	return os.Getenv("WOODPECKER_SSH_KEY") != "" || os.Getenv("WOODPECKER_SSH_PASSWORD") != ""
+}
+
+func (e *ssh) Load() error {
+	cmd, err := e.client.Command("/bin/env", "mktemp", "-d", "-p", "/tmp", "woodpecker-ssh-XXXXXXXXXX")
```

Comment:
@qwerty287 Am I right that the code would currently create a `/tmp/woodpecker-ssh-XXXXXXXXXX` folder once and place one folder per pipeline with the name of the pipeline inside of it?